### PR TITLE
Build the backend site in buildspec.yml

### PR DESCRIPTION
**Pull request by glgh**
# Pull Request Template

## Issue Overview

This PR addresses metrotranscom/doorway-infra#64
Note that this PR probably won't work yet. Expecting a follow-up CL in the infra repo (once this PR is merged and can be tested with). See notes below.

- [ ] This change addresses the issue in full
- [x] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Build the backend Docker image and push it to ECR

## How Can This Be Tested/Reviewed?

This has been tested locally with Codebuild local script, with some environmental variables manually fed in. It may not work out of the box in the actual CI/CD pipeline, and I expect a follow up CL in the doorway-infra repo to update the terraform configs.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by isaac-looker** [comment]
Can we add documentation on how to invoke this? Is the plan to _only_ allow terraform to invoke this?

File: buildspec.yml [hunk 1]
```diff
@@ -1,10 +1,30 @@
...
 
-# For now, this is just a placeholder to test the CodeBuild setup
+# Builds the backend site with Docker and the image to ECR. 
+# TODO: add public and private sites.
 
 version: 0.2
 
+env:
+  variables:
+    IMAGE_TAG: "latest-test" # TODO: change to latest?
```
Comment (chriscasto) [inline buildspec.yml:10]: We'll actually have at least a couple of images per service.  One will be set up for testing and another will be optimized for production.  They will use the same Dockerfile but have different targets.  I've been using "test" and "run" tags for those, respectively, but that's not a hard requirement.  We'll actually probably want to use hash-based tags until we're ready to promote as part of the final deployment step, but again, that can easily be added later.

**Review by chriscasto — COMMENTED** [review]
Adding as a comment for now since this file is just for testing but I want to call out a couple of things in addition to my inline comments.  Just give me an ack once you've had a chance to read through.

Since this is a monorepo with multiple services, we'll need multiple buildspec files, one for each action (build, test) for each service/task and then a final deploy action to update the compute resources for all of them at the same time, assuming everything else is green.  We'll also want to put these in the "build" directory to keep all of the build configs together in one place.

**Comment by chriscasto** [comment]
GuangHui has left and this work has been implemented in subsequent PRs.  Closing

**Pull request closed without merging**